src/features/filter.ts

```diff
@@ -1,12 +1,50 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { Status } from '../types/Status';
 
-const initialState = {
+type SetQueryAction = {
+  type: 'filter/SET_QUERY';
+  payload: string;
+};
+
+type SetStatusAction = {
+  type: 'filter/SET_STATUS';
+  payload: Status;
+};
+
+const setQuery = (query: string): SetQueryAction => ({
+  type: 'filter/SET_QUERY',
+  payload: query,
+});
+
+const setStatus = (status: Status): SetStatusAction => ({
+  type: 'filter/SET_STATUS',
+  payload: status,
+});
+
+export const actions = { setQuery, setStatus };
+
+type State = {
+  query: string;
+  status: Status;
+};
+
+const initialState: State = {
   query: '',
   status: 'all',
 };
 
-export const filterSlice = createSlice({
-  name: 'filter',
-  initialState,
-  reducers: {},
-});
+type Action = SetQueryAction | SetStatusAction;
+
+const filterReducer = (action: Action, state: State = initialState): State => {
```

Comment:
The `filterReducer` function has its parameters in the wrong order. According to Redux convention, the first parameter should be `state` and the second should be `action`. The correct signature should be `const filterReducer = (state: State = initialState, action: Action): State => {`.